src/AzureRenderManager/WebApp/Controllers/ReportingController.cs

```diff
@@ -5,25 +5,59 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
+using TaskTupleAwaiter;
 using WebApp.Code.Attributes;
 using WebApp.Code.Contract;
+using WebApp.Config;
+using WebApp.CostManagement;
 
 namespace WebApp.Controllers
 {
     [MenuActionFilter]
     [EnvironmentsActionFilter]
     public class ReportingController : MenuBaseController
     {
+        private readonly CostManagementClientAccessor _clientAccessor;
+
         public ReportingController(
             IEnvironmentCoordinator environmentCoordinator,
             IPackageCoordinator packageCoordinator,
-            IAssetRepoCoordinator assetRepoCoordinator) :
-            base(environmentCoordinator, packageCoordinator, assetRepoCoordinator)
+            IAssetRepoCoordinator assetRepoCoordinator,
+            CostManagementClientAccessor clientAccessor)
+            : base(environmentCoordinator, packageCoordinator, assetRepoCoordinator)
         {
+            _clientAccessor = clientAccessor;
         }
 
         [HttpGet]
         [Route("Reporting")]
-        public IActionResult Index() => View();
+        public async Task<ActionResult> Index()
+        {
+            var (envs, client) = await (Environments(), _clientAccessor.GetClient());
+
+            var usages = await Task.WhenAll(envs.Select(env => GetUsage(client, env)));
+
+            return View(usages);
+        }
+
+        private static async Task<(string envName, UsageResponse)> GetUsage(CostManagementClient client, RenderingEnvironment env)
+        {
+            var usageRequest =
+                new UsageRequest(
+                    Timeframe.MonthToDate,
+                    new Dataset(
+                        Granularity.Daily,
+                        new Dictionary<string, Aggregation>
```

Comment:
We should support daily costs in addition to aggregate costs. (can add it later)

Reply:
This still produces daily costs, the aggregation is needed to produce any outputs at all.

Reply:
I'm guessing that the cost for a day is made up of many charges so you have to aggregate them.